diff -NarU 20 a/modules/features2d/src/sift.cpp b/modules/features2d/src/sift.cpp
--- a/modules/features2d/src/sift.cpp	2010-12-05 14:35:24.000000000 +1100
+++ b/modules/features2d/src/sift.cpp	2011-05-14 10:57:27.435726626 +1000
@@ -31,44 +31,40 @@
 // This software embodies a method for which the following patent has
 // been issued: "Method and apparatus for identifying scale invariant
 // features in an image and use of same for locating an object in an
 // image," David G. Lowe, US Patent 6,711,293 (March 23,
 // 2004). Provisional application filed March 8, 1999. Asignee: The
 // University of British Columbia.
 //
 // IN NO EVENT SHALL THE UNIVERSITY OF CALIFORNIA BE LIABLE TO ANY PARTY
 // FOR DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES,
 // INCLUDING LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND
 // ITS DOCUMENTATION, EVEN IF THE UNIVERSITY OF CALIFORNIA HAS BEEN
 // ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THE UNIVERSITY OF
 // CALIFORNIA SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT
 // LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
 // A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS ON AN "AS IS"
 // BASIS, AND THE UNIVERSITY OF CALIFORNIA HAS NO OBLIGATIONS TO PROVIDE
 // MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
 
 #include "precomp.hpp"
 
-#ifdef __arm__
-#define ARM_NO_SIFT
-#endif
-
 #ifdef ANDROID
 #undef  ARM_NO_SIFT
 #endif //ANDROID
 
 #ifndef ARM_NO_SIFT
 
 #include <iostream>
 #include <limits>
 
 #define log2(a) (log((a))/CV_LOG2)
 
 #if defined _MSC_VER && _MSC_VER >= 1400
 #pragma warning(disable: 4100 4244 4267 4305)
 #endif
 
 /*
  * from sift.hpp of original code
  */
 #if defined (VL_USEFASTMATH)
 #if defined (VL_MAC)
