diff -Nuar a/config.mk b/config.mk
--- a/config.mk	2011-04-09 00:08:24.000000000 +1000
+++ b/config.mk	2011-06-05 16:10:31.726553065 +1000
@@ -116,7 +116,7 @@
 gccincdir := $(shell $(CC) -print-file-name=include)
 
 CPPFLAGS := $(DBGFLAGS) $(OPTFLAGS) $(RELFLAGS)		\
-	-D__KERNEL__ -DTEXT_BASE=$(TEXT_BASE)		\
+	-D__KERNEL__ -DTEXT_BASE=$(TEXT_BASE) -D__EXTRAVERSION__=\"$(EXTRAVERSION)\" \
 	-I$(TOPDIR)/include				\
 	-fno-builtin -ffreestanding -nostdinc -isystem	\
 	$(gccincdir) -pipe $(PLATFORM_CPPFLAGS)
diff -Nuar a/lib/board.c b/lib/board.c
--- a/lib/board.c	2011-06-05 10:41:18.000000000 +1000
+++ b/lib/board.c	2011-06-05 12:26:52.455256139 +1000
@@ -40,8 +40,8 @@
 #ifdef CFG_PRINTF
 int print_info(void)
 {
-	printf("\n\nTexas Instruments X-Loader 1.51 ("
-			__DATE__ " - " __TIME__ ")\n");
+	printf("\n\nTexas Instruments X-Loader 1.51" 
+			__EXTRAVERSION__ " (" __DATE__ " - " __TIME__ ")\n");
 	return 0;
 }
 #endif
